lib/cylc/batch_sys_manager.py

```diff
@@ -695,6 +700,10 @@ def _job_submit_impl(
             # batch_sys.submit should handle OSError, if relevant.
             ret_code, out, err = batch_sys.submit(job_file_path)
         else:
+            env = None
+            if hasattr(batch_sys, "SUBMIT_CMD_ENV"):
+                env = dict(os.environ)
```

Comment:
`Popen(env=environ)` appears to execute the command with *only* the environment variables specified in `environ`, so we need to pass `os.environ` *plus* those in `SUBMIT_CMD_ENV` to it. Otherwise, our command will be running with nothing but `SHELL=sh`.